### ISSUE TITLE
Manual Reload of Integration required

### ISSUE THREAD

**Issue by kellerassel007** [petergridge/openweathermaphistory#42]
Last night I noticed an outage of the sensor data. It required a reload of the sensor settings and then it worked again instantly.
The days before it worked without any outage, any clues?

Debug log:  
2024-07-25 11:12:55.393 ERROR (MainThread) [custom_components.openweathermaphistory.sensor] Unexpected error fetching openweathermaphistory data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 312, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/openweathermaphistory/sensor.py", line 107, in _async_update_data
    await self._weather.async_update()
  File "/config/custom_components/openweathermaphistory/weatherhistory.py", line 308, in async_update
    earliestdata = min(data)
                   ^^^^^^^^^
TypeError: 'NoneType' object is not iterable

2024-07-25 11:22:55.392 ERROR (MainThread) [custom_components.openweathermaphistory.sensor] Unexpected error fetching openweathermaphistory data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 312, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/openweathermaphistory/sensor.py", line 107, in _async_update_data
    await self._weather.async_update()
  File "/config/custom_components/openweathermaphistory/weatherhistory.py", line 308, in async_update
    earliestdata = min(data)
                   ^^^^^^^^^
TypeError: 'NoneType' object is not iterable

2024-07-25 11:32:55.392 ERROR (MainThread) [custom_components.openweathermaphistory.sensor] Unexpected error fetching openweathermaphistory data
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 312, in _async_refresh
    self.data = await self._async_update_data()
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/openweathermaphistory/sensor.py", line 107, in _async_update_data
    await self._weather.async_update()
  File "/config/custom_components/openweathermaphistory/weatherhistory.py", line 308, in async_update
    earliestdata = min(data)
                   ^^^^^^^^^
TypeError: 'NoneType' object is not iterable

2024-07-25 11:41:42.077 DEBUG (MainThread) [custom_components.openweathermaphistory.data] Updating from https://api.openweathermap.org/data/3.0/onecall/timemachine?
[...]

worked from here again after reload of sensor settings

**Comment by kellerassel007** [comment]
My workaround is to build an automation that monitors my sensor and trigger reload in case the sensor is unavailable for at least 1 hour:
```yaml
trigger:
  - platform: state
    entity_id:
      - sensor.<YOUR_SENSOR_ENTITY_ID>
    to: unavailable
    for:
      hours: 1
      minutes: 0
      seconds: 0
  - platform: state
    entity_id:
      - sensor.<YOUR_SENSOR_ENTITY_ID>
    to: unknown
    for:
      hours: 1
      minutes: 0
      seconds: 0
condition: []
action:
  - service: homeassistant.reload_config_entry
    data:
      entry_id: "{{ config_entry_id('sensor.<YOUR_SENSOR_ENTITY_ID>') }}"
```

**Comment by oldgitdaddy** [comment]
Same error here. Can someone please look into this?

**Comment by petergridge** [comment]
Hi can you let me know the following:
- HomeAssistant version
- OpenWeatherMapHistory Version
- What operating system and hardware you are using
I can't replicate the problem on my systems: 
- RapberryPI4 and HA OS 13.0 and 2024.8 Home assistant
- Dev environment Docker/Windows

I am using the latest BETA 2.0.14 of OWM

**Comment by kellerassel007** [comment]
> Hi can you let me know the following:
> 
>     * HomeAssistant version
> 
>     * OpenWeatherMapHistory Version
> 
>     * What operating system and hardware you are using
> 
> 
> I can't replicate the problem on my systems:
> 
>     * RapberryPI4 and HA OS 13.0 and 2024.8 Home assistant
> 
>     * Dev environment Docker/Windows
> 
> 
> I am using the latest BETA 2.0.14 of OWM

Thanks for your support!
This is a rare error, which only happened once for me in the last 2 months since I use this integration.

HomeAssistant HA OS (running virtualized on Proxmox host):
Core 2024.7.0
Supervisor 2024.08.0
Operating System 12.4
Frontend 20240703.0

OpenWeatherMapHistory: 2.0.12

**Comment by petergridge** [comment]
I think I have found the issue here, the refresh rate has been published with my testing frequency. I will release a version with the normal 30 minute refresh (OWM only updated once an hour). 

The only time I had a similar issue is when I exceed the allocated calls. Hopefully this change resolves the immediate issue. I also need to update the code to handle the situation more cleanly, maybe add a status value that can be displayed as a sensor.

I will publish the new version in the next day or so.
